pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -479,6 +480,51 @@ def _build_circbox(instr: Instruction, circuit: QuantumCircuit) -> CircBox:
     return CircBox(subc)
 
 
+def _pytket_boxes_from_IfElseOp(
+    if_else_op: IfElseOp, qregs: list[QuantumRegister], cregs: list[ClassicalRegister]
+) -> tuple[CircBox, CircBox]:
+    if_qc: QuantumCircuit = if_else_op.params[0]
+    else_qc: QuantumCircuit = if_else_op.params[1]
+
+    if_builder = CircuitBuilder(qregs, cregs)
+    if_builder.add_qiskit_data(if_qc)
+    if_circuit = if_builder.circuit()
+    if_circuit.name = "If"
+
+    else_builder = CircuitBuilder(qregs, cregs)
+    else_builder.add_qiskit_data(else_qc)
+    else_circuit = else_builder.circuit()
+    else_circuit.name = "Else"
+
+    return CircBox(if_circuit), CircBox(else_circuit)
+
+
+def build_if_else_circuit(
+    if_else_op: IfElseOp,
+    qregs: list[QuantumRegister],
+    cregs: list[ClassicalRegister],
+    qubits: list[Qubit],
+    bits: list[Bit],
+) -> Circuit:
+    if_box, else_box = _pytket_boxes_from_IfElseOp(if_else_op, qregs, cregs)
+    circ_builder = CircuitBuilder(qregs, cregs)
+    circ = circ_builder.circuit()
```

Comment:
```
print(circ.qubits, circ.bits)
print(if_box.get_circuit().qubits, if_box.get_circuit().bits)
print(qubits)
print(bits)
```
prints
```
[q105[0], q105[1]] [c3[0], c3[1]]
[q105[0], q105[1]] [c3[0], c3[1]]
[q105[1]]
[c3[0]]
```

So the box has 2 qubits and 2 bits, but in `add_circbox`, `args=qubits + bits` only provides 1 qubit and 1 bit.

Reply:
Thanks. 

I figured out what I was doing wrong. The thing that was confusing me was that the two circuits which define the `IfElseOp` do not have any registers. Their qubits are tied to the register of the circuit containing the `IfElseOp` this meant that the subcircuits in my `CircBox`es had too many qubits/bits as there were some additional blank wires.

Cleaning up the blank wires as in https://github.com/CQCL/pytket-qiskit/pull/437/commits/d65a347292a0238499dde5a10abb6442543ec745 seems to fix this.